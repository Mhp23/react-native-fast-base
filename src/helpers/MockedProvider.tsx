import React from 'react';
import {DefaultTheme, ThemeContext, type ThemeContextProps} from '../core';

const MockedProvider: React.FC = ({children}) => {
  return (
    <ThemeContext.Provider
      value={{
        ...({} as ThemeContextProps),
        ...DefaultTheme.DefaultTheme,
      }}>
      {children}
    </ThemeContext.Provider>
  );
};

export default MockedProvider;
