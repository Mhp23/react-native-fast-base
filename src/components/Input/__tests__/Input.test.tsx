import React from 'react';
import MockedProvider from '../../../helpers/MockedProvider';
import {render, cleanup} from '@testing-library/react-native';
import OutlineInput from '../Outline';
import UnderlineInput from '../Underline';

afterEach(cleanup);

describe('Input components', () => {
  it('should render outline correctly', () => {
    const rendered = render(
      <MockedProvider>
        <OutlineInput />
      </MockedProvider>,
    );
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });

  it('should render underline correctly', () => {
    const rendered = render(
      <MockedProvider>
        <UnderlineInput />
      </MockedProvider>,
    );
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
});
