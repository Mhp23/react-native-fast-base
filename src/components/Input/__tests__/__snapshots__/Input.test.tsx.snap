// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input components should render outline correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "transform": Array [
        Object {
          "translateX": 0,
        },
        Object {
          "translateY": 0,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "borderColor": "#DDDDDD",
        "borderRadius": 8.333333333333334,
        "borderWidth": 3.125,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": 16.666666666666668,
        "overflow": "hidden",
        "position": "relative",
      }
    }
    testID="FAST_BASE_WRAPPER"
  >
    <TextInput
      accessible={true}
      editable={true}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#626567"
      secureTextEntry={false}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 27.083333333333336,
          "paddingBottom": 21.666666666666668,
          "paddingEnd": 27.083333333333336,
          "paddingStart": 27.083333333333336,
          "paddingTop": 21.666666666666668,
        }
      }
    />
  </View>
</View>
`;

exports[`Input components should render underline correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "transform": Array [
        Object {
          "translateX": 0,
        },
        Object {
          "translateY": 0,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "borderBottomColor": "#DDDDDD",
        "borderBottomWidth": 3.125,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": 16.666666666666668,
        "position": "relative",
      }
    }
    testID="FAST_BASE_WRAPPER"
  >
    <TextInput
      accessible={true}
      editable={true}
      onBlur={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#626567"
      secureTextEntry={false}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 27.083333333333336,
          "paddingBottom": 21.666666666666668,
          "paddingTop": 21.666666666666668,
        }
      }
    />
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#3b82f6",
          "bottom": -3.125,
          "height": 3.125,
          "position": "absolute",
          "transform": Array [
            Object {
              "scaleX": 0,
            },
          ],
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
